docs: use_render_queue, use_liveness_scope, use_table_listener docs (#1044)
deephaven/deephaven-plugins: plugins/ui/docs/hooks/use_liveness_scope.md

# use_liveness_scope

> `use_liveness_scope` allows your to interact with the [liveness scope](https://deephaven.io/core/docs/conceptual/liveness-scope-concept/) for a component. Some functions which interact with a component will create live objects that need to be managed by the component to ensure they are kept active.

Reviewer: ```suggestion
`use_liveness_scope` allows you to interact with the [liveness scope](https://deephaven.io/core/docs/conceptual/liveness-scope-concept/) to manage live objects within a component. Some functions that interact with a component will create live objects that need to be managed by the component to ensure they are kept active.
```

Reviewer: Not 100% sure of accuracy, but the sentence needed help.

> `use_liveness_scope` allows you to interact with the [liveness scope](https://deephaven.io/core/docs/conceptual/liveness-scope-concept/) to manage live objects within a component. Some functions that interact with a component will create live objects that need to be managed by the component to ensure they are kept active.

Reviewer 2: I've reviewed other PRs that use the absolute linking syntax as if linking from enterprise to core:

```
/core/docs/conceptual/liveness-scope-concept/
```

This is better for now because the link will actually work, but it's also something that might have to change once Salmon is live.

> The primary use case for this is when creating tables outside the component's own function, and passing them as state for the component's next update. If the table is not kept alive by the component, it will be garbage collected and the component will not be able to update with the new data.

Reviewer: ```suggestion
The primary use case is when creating tables outside the component's own function, and passing them as state for the component's next update. If the table is not kept alive by the component, it will be garbage collected and the component will not be able to update with the new data.
```

> The primary use case is when creating tables outside the component's own function, and passing them as state for the component's next update. If the table is not kept alive by the component, it will be garbage collected and the component will not be able to update with the new data.

Reviewer 2: ```suggestion
The primary use case is to create tables outside the component's own function, and pass them as state for the component's next update. If the table is not kept alive by the component, it will be garbage collected and the component will not be able to update with the new data.
```

## Example

```python
from deephaven import ui, time_table

@ui.component
def ui_resetable_table():
    table, set_table = ui.use_state(lambda: time_table("PT1s"))
    handle_press = ui.use_liveness_scope(lambda _: set_table(time_table("PT1s")), [])
    return [
        ui.action_button(
            "Reset",
            on_press=handle_press,
        ),
        table,
    ]

resetable_table = ui_resetable_table()
```

## UI recommendations

> 1. **Avoid using `use_liveness_scope` unless necessary**: This is an advanced feature that should only be used when you need to manage the liveness of objects outside of the component's own function. Prefer instead to derive a live component based on state rather than setting a live component within state.

Reviewer: ```suggestion
1. **Avoid using `use_liveness_scope` unless necessary**: This is an advanced feature that should only be used when you need to manage the liveness of objects outside of the component's own function. Instead, derive a live component based on state rather than setting a live component within state.
```

2. **Use `use_liveness_scope` to manage live objects**: If you need to manage the liveness of objects created outside of the component's own function, use `use_liveness_scope` to ensure they are kept alive. For more information on liveness scopes and why they are needed, see the [liveness scope documentation](https://deephaven.io/core/docs/conceptual/liveness-scope-concept/).

> ## Refactoring to avoid liveness scope

Reviewer 2: No `ing` in headings.
